### ISSUE TITLE
User Feedback

### ISSUE THREAD

**Issue by outoftime** [popcodeorg/popcode#965]
Submitter: kathleen
Email: kcoen29@gmail.om
i am very angry with popcode. i will probably never use it again. why the hell dont you have a save button. this is bs. i have to just "trust" your dumbass website and today I lost hours worth of code. go to hell

**Comment by tenyoung795** [comment]
I wonder if her internet went out during her session and so the saves would fail, but the UI never reported these failures?

At first, I was worried #855 may be the culprit: would [`throttle`](https://github.com/redux-saga/redux-saga/tree/master/docs/api#throttlems-pattern-saga-args) ever wait for its last task to finish before starting a new task? IIUC the documentation though, it only waits since the last task *spawn*, not the last task termination.

Also, although [`clients/firebase/saveCurrentProject`](https://github.com/popcodeorg/popcode/blob/39f40d0b0ffbe19ffa4eba219d23a6a5645dd9dd/src/clients/firebase.js#L76) returns a promise, [`util/projectUtils/saveCurrentProject`](https://github.com/popcodeorg/popcode/blob/39f40d0b0ffbe19ffa4eba219d23a6a5645dd9dd/src/util/projectUtils.js#L23) lets it go out of scope without handling any errors. Seems like that needs to be changed so that Popcode can at least report save errors to the UI, if it doesn't already?

**Comment by outoftime** [comment]
@tenyoung795 I reached directly out to the user so I will update this issue if I hear back. All your theories are good ones, and that’s a very good point about handling save errors (want to create a separate issue for that?)

However my best guess is that the user simply did not log in, and probably did not realize that was an option; so, when she closed the environment, her work was indeed lost.

**Comment by outoftime** [comment]
I didn’t receive a response from the person who reported this bug so I’m going to go ahead and close it. There [is now an issue capturing Firebase errors](https://github.com/popcodeorg/popcode/issues/1009).
